src/main/kotlin/fi/metatavu/muisti/persistence/dao/AbstractDAO.kt

```diff
@@ -79,9 +76,9 @@ abstract class AbstractDAO<T> {
     open fun listAll(firstResult: Int, maxResults: Int): List<T> {
         val genericTypeClass: Class<*>? = genericTypeClass
         val query: Query = entityManager.createQuery("select o from " + genericTypeClass!!.name + " o")
-        query.firstResult = firstResult
-        query.maxResults = maxResults
-        return query.resultList as List<T>
+        query.setFirstResult(firstResult)
+        query.setMaxResults(maxResults)
+        return query.getResultList() as List<T>
```

Comment:
it's simply revert to the original version of the class since Leppä was wondering why whole class was replaced. But you are correct, it could be cleaned up